tests/test_views.py

```diff
@@ -0,0 +1,108 @@
+"""Tests for views are at this file."""
+from unittest.mock import patch
+from datetime import datetime
+from http.client import OK as HTTP_OK
+from django.test import TestCase, Client
+from django.contrib.auth.models import User
+from django.urls import reverse
+from django.utils import timezone
+from blog.models import Post
+
+
+class ViewsTest(TestCase):
+    """Testing class for views."""
+
+    USERNAME = 'admin'
+    PASSWORD = '12345678'
+
+    def setUp(self):
+        """Prepare data for testing."""
+        self.client = Client()
+        self.user = User.objects.create(
+            username='testuser',
+            password='blablabla',
+            is_superuser=True, is_staff=True,
+            email='testuser@gmail.com',
+            is_active=True,
+        )
+
+    def test_index_view_rendering(self):
+        """Testing main page for render needed posts."""
+        time_zone = timezone.get_current_timezone()
+        future_post = Post.objects.create(
+            author=self.user,
+            title='future_test',
+            text='superText',
+            created_date=datetime(
+                day=1, month=4,
+                year=3020,
+                tzinfo=time_zone,
+            ),
+            published_date=datetime(
+                day=1,
+                month=4,
+                year=3020,
+                tzinfo=time_zone,
+            ),
+        )
+        post = Post.objects.create(
+            author=self.user, title='Test',
+            text='superText',
+            created_date=datetime(
+                day=1, month=3,
+                year=2020, tzinfo=time_zone,
+            ),
+            published_date=datetime(
+                day=1, month=3,
+                year=2020,
+                tzinfo=time_zone,
+            ),
+        )
+        past_post = Post.objects.create(
+            author=self.user, title='past_est',
+            text='superText',
+            created_date=datetime(
+                day=1, month=4,
+                year=2019,
+                tzinfo=time_zone,
+            ),
+            published_date=datetime(
+                day=1, month=4,
+                year=2019,
+                tzinfo=time_zone,
+            ),
+        )
+
+        post_list_url = reverse('post_list')
+        posts_at_date = {
+            # long time ago
+            datetime(day=1, month=4, year=3020, tzinfo=time_zone): [],
```

Comment:
3020 is in the future but the comment says it's a long time ago. It should be a date before the first post.

Reply:
@igorkruglyak ^